docroot/sites/all/modules/features/ilr_sdc_listings/ilr_sdc_listings.module

```diff
@@ -2600,7 +2600,10 @@ function _ilr_sdc_listings_get_class_id($wrapper) {
 function ilr_sdc_listings_node_insert($node) {
   if ($node->type == 'sdc_class') {
     if (empty($node->field_registration_end_date)) {
-      $wrapper = ilr_get_node_wrapper($node);
+      // Reload the node so that we can wrap it and save it
+      // This prevents the error: `Integrity constraint violation: 1062 Duplicate entry`
+      $loaded_node = node_load($node->nid);
+      $wrapper = ilr_get_node_wrapper($loaded_node);
       $start_date = $wrapper->field_class_dates->value()['value'];
```

Comment:
As discussed, we should try moving this to `hook_node_presave()` so that the node isn't saved twice.

And if we want to dodge the overhead of an entity metadata wrapper, here's some info on updating Drupal date fields directly: https://drupal.stackexchange.com/questions/26948/programmatically-set-date-fields-in-drupal-7-date-date-iso-format-and-date

Reply:
Ok. I did a fixup. However, it strikes me that it might not be ideal to run a fixup during a pr. Let's discuss. At any rate, this is working for me.